back/engines/commercial/idea_custom_fields/app/controllers/idea_custom_fields/web_api/v1/admin/idea_custom_fields_controller.rb

```diff
@@ -104,26 +104,87 @@ def raise_error_if_not_geographic_field
       raise "Custom field with input_type: '#{@custom_field.input_type}' is not a geographic type"
     end
 
+    def validate!
+      fields = update_all_params.fetch(:custom_fields).map do |field|
+        CustomField.new(field.slice(:code, :key, :input_type, :title_multiloc, :description_multiloc, :required, :enabled, :ordering))
+      end
+      validate_non_empty_form!(fields)
+      validate_stale_form_data!(fields)
+      validate_end_page!(fields)
+      validate_first_page!(fields)
+      validate_separate_title_body_pages!(fields)
+    end
+
+    def validate_non_empty_form!(fields)
+      return if !fields.empty?
+
+      raise UpdateAllFailedError, { form: [{ error: 'empty' }] }
+    end
+
     # To try and avoid forms being overwritten with stale data, we check if the form has been updated since the form editor last loaded it
     # But ONLY if the FE sends the form_last_updated_at param
-    def raise_error_if_stale_form_data
+    def validate_stale_form_data!(_)
```

Comment:
Why pass in fields `_` if they're not needed?

Reply:
I think consistency was the reasoning behind this, but makes sense to remove it, I'll do that.